Sources/StreamChatUI/ChatMessageList/RETHINK/DefaultMessageLayoutOptionsResolver.swift

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+
+public typealias _LayoutOptionsResolver<ExtraData: ExtraDataTypes> =
+    (_ message: _ChatMessage<ExtraData>, _ isLastInGroup: Bool) -> ChatMessageLayoutOptions
```

Comment:
I think this is too restricting. The "group" is what should happen inside the resolver. Maybe the API can rather look something like this:
```swift
public typealias MessageLayoutOptionsResolver = (_ indexPath: IndexPath, _ message: (IndexPath) -> Message?) -> ChatMessageLayoutOptions
```
which would make it possible to access all messages you need.

The default resolver could look like:
```swift
 func DefaultMessageLayoutOptionsResolver(
    minTimeInteravlBetweenMessagesInGroup: TimeInterval = 10
) -> MessageLayoutOptionsResolver
```

Reply:
updated ✅